src/dsql/ExprNodes.cpp

```diff
@@ -194,6 +192,55 @@ static void setParameterInfo(dsql_par* parameter, const dsql_ctx* context);
 //--------------------
 
 
+int SortValueItem::compare(const dsc* desc1, const dsc* desc2)
+{
+	return MOV_compare(JRD_get_thread_data(), desc1, desc2);
+}
+
+LookupValueList::LookupValueList(MemoryPool& pool, const ValueListNode* values, ULONG impure)
+	: m_values(pool, values->items.getCount()), m_impureOffset(impure)
+{
+	for (const auto item : values->items)
+		m_values.add(const_cast<ValueExprNode*>(item.getObject()));
+}
+
+TriState LookupValueList::find(thread_db* tdbb, Request* request, const ValueExprNode* value, const dsc* desc) const
+{
+	const auto impure = request->getImpure<impure_value>(m_impureOffset);
+	auto sortedList = impure->vlu_misc.vlu_sortedList;
+
+	if (!(impure->vlu_flags & VLU_computed))
+	{
+		delete impure->vlu_misc.vlu_sortedList;
```

Comment:
It is good to initialize the field to `nullptr` just after the delete.
Because if the new constructor fail, it could be left with garbage.
This is done after `delete impure->vlu_misc.vlu_invariant;` in `SubstringSimilarNode::execute`.

Reply:
Agreed.